src/node_crypto.cc

```diff
@@ -5263,7 +5263,7 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_IF_NOT_BUFFER(args[1], "Salt");
 
   pass = static_cast<char*>(node::Malloc(passlen));
-  if (pass == nullptr) {
+  if (pass == nullptr && passlen > 0) {
     FatalError("node::PBKDF2()", "Out of Memory");
   }
```

Comment:
@mscdex The below block seems to be more clear as you concerned.
```cpp
if (passlen > 0) {
  pass = static_cast<char*>(node::Malloc(passlen));
  if (pass == nullptr) {
    FatalError("node::PBKDF2()", "Out of Memory");
  }
}
```

And for this question,

> At any rate, I don't see anything wrong in doing it the way it's currently being done.

Yes, the current being done is not wrong in anything, this is just a suggestion, when `passlen` is 0 or less than that, we could decrease calls to `node::Malloc()` by what I did propose :)

But I'm not sure if the compiler will optimize this or this is valuable to make this change.